src/backend/distributed/commands/database.c

```diff
@@ -185,17 +201,33 @@ PreprocessAlterDatabaseStmt(Node *node, const char *queryString,
 		return NIL;
 	}
 
-	AlterDatabaseStmt *stmt = castNode(AlterDatabaseStmt, node);
-
 	EnsureCoordinator();
 
+	AlterDatabaseStmt *stmt = castNode(AlterDatabaseStmt, node);
+
 	char *sql = DeparseTreeNode((Node *) stmt);
 
-	List *commands = list_make3(DISABLE_DDL_PROPAGATION,
-								(void *) sql,
-								ENABLE_DDL_PROPAGATION);
+	List *commands = NULL;
 
-	return NodeDDLTaskList(NON_COORDINATOR_NODES, commands);
+	if (isSetTablespaceStatement(stmt))
+	{
+		/*Set tablespace does not work inside a transaction.Therefore, we close the transaction before set tablespace
+		 * and open it again after set tablespace.
+		 */
```

Comment:
not applicable anymore

Reply:
Fixed the comment. Organized the code to be more concise